src/ncp/ncp_base_ftd.cpp

```diff
@@ -378,11 +379,12 @@ otError NcpBase::InsertPropertyHandler_THREAD_JOINERS(const uint8_t *aValuePtr,
                        aValuePtr,
                        aValueLen,
                        (
-                           SPINEL_DATATYPE_UTF8_S     // PSK
-                           SPINEL_DATATYPE_UINT32_S   // Timeout
-                           SPINEL_DATATYPE_EUI64_S    // Extended address
+                           SPINEL_DATATYPE_DATA_WLEN_S  // PSK
+                           SPINEL_DATATYPE_UINT32_S     // Timeout
+                           SPINEL_DATATYPE_EUI64_S      // Extended address
                        ),
-                       &aPSKd,
+                       &pskd,
+                       &pskdLength,
```

Comment:
I certainly agree that using encoding of  `DATA_WLEN` makes more sense here. I also appreciate that you made the corresponding changes/PR in `wpantund` and also updated the spinel documentation. 

This PR will change an existing Spinel property definition/format. Usually we only consider appending new elements/fields to end of an existing format (to keep it backward compatible) or we try to define a new property if the format needs to change completely (e.g., we ensured that the whitelist/blacklist spinel properties behave as before and added new capabilities through extensions/new properties when the OT MAC filter API got changed). 

That said, since this is relatively new definition, I think we can still change it. 

I think @pszkotak (and the Nordic team) contributed the original code for joiner/commissioner properties in 2fa7a13c1b59057ee30fd1f9bc8f358344df3dc9.  @pszkotak are you ok with this change? Thanks.

Reply:
I was hesitant to do the changes in the spinel (as can be seen in https://github.com/openthread/openthread/pull/2185#issuecomment-331646223), but jwhui seemed to agree with the proposed change. If there was a strong reason against, then it would not be a big problem to revert before the 6a67899 commit. In any case, I hope it's acceptable.

Reply:
@abtink, changing the PROP_THREAD_JOINERS to "A(t(ULE))" to "A(t(dLE))" will break the backward compability, but if we make sure that wpantund gets aligned it shouldn't be a problem IMHO.
Passing the DATATYPE_DATA_WLEN seems neater than DATATYPE_UTF8 that we have now.

Could you confirm at which point would you like to perform the base32 conversion?
Is it going to take place before the wpantund sends it via spinel (less data would be sent, which seems like a good idea) or afterwards on the NCP itself?

Reply:
Thanks @pszkotak . So looks like we all agree to allow the spinel property format change. Let's also merge the wpantund PR https://github.com/openthread/wpantund/pull/242 at same time...

Reply:
I remain unconvinced, it has not been made clear to me (maybe I missed it) why this is a good idea.